include/CFinalBoneHierarchy.h

```diff
@@ -101,6 +101,42 @@ namespace scene
                 createAnimationKeys(inLevelFixedJoints);
             }
 
+			//! Constructor from blob.
+			/**It's the only class in which such constructor exists since a lot of private members are copied with just `memcpy` call. Blob's data should be filled by fillExportBlob().
+			Meant to be used by loaders only.
+			@see @ref CBAWMeshFileLoader CBAWMeshWriter
+			*/
+			explicit CFinalBoneHierarchy(typename core::CorrespondingBlobTypeFor<CFinalBoneHierarchy>::type* _blob) :
```

Comment:
I don't like this.

The blobs should create a CFinalBoneHierarchy, not the CFBH from Blob.

The reason is that imagine having 4 versions of the blob and also updating the CFBH class.

If you want the loader to be backward compatible with all Blob versions 1 through 4, then you will have 4 constructors polluting this file which should only be about getting interpolated keyframes for an armature.

Instead have a constructor:
`explicit CFinalBoneHierarchy(BoneReferenceData* boneFlatArrayBegin, BoneReferenceData* boneFlatArrayEnd, ..., etc.)`

Then when we move to C++11 we can use move semantics to move the bone name strings.